src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -80,19 +81,55 @@ func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 
 	r.LastObservedState = &sts
 
-	// Ensure StatefulSet #replicas equals cluster requirement.
-	if sts.Spec.Replicas != nil && r.pandaCluster.Spec.Replicas != nil && *sts.Spec.Replicas != *r.pandaCluster.Spec.Replicas {
-		r.logger.Info(fmt.Sprintf("StatefulSet %s has replicas set to %d but need %d. Going to update", r.Key().Name, *sts.Spec.Replicas, *r.pandaCluster.Spec.Replicas))
-
-		sts.Spec.Replicas = r.pandaCluster.Spec.Replicas
+	updated := Update(&sts, r.pandaCluster, r.logger)
+	if updated {
```

Comment:
```suggestion
	if r.isResourceDifferent() || r.isReplicasDifferent() {
```

Reply:
hmm I actually prefer it like this, then the internals of what is updated is hidden in one method that can be also tested as a whole piece.

I think in the end we won't even be distinguishing between what gets updated - then we would end up with isXUpdated() || isYUpdated() || isAUpdated() etc. while all you care in this level I think is whether the resource was updated ... in any way

Reply:
Ok. I would only inline the function call. IMO this should be private function that is tested via `Ensure` function.